extensions/sql-migration/src/api/utils.ts

```diff
@@ -600,6 +600,53 @@ export async function getVirtualMachines(account?: Account, subscription?: azure
 	return virtualMachines;
 }
 
+export async function getVirtualMachinesDropdownValues(virtualMachines: azure.SqlVMServer[], location: azureResource.AzureLocation, resourceGroup: azureResource.AzureResourceResourceGroup, account: Account, subscription: azureResource.AzureResourceSubscription): Promise<CategoryValue[]> {
+	let virtualMachinesValues: CategoryValue[] = [];
+	if (location && resourceGroup) {
+		for (const virtualMachine of virtualMachines) {
+			if (virtualMachine.location.toLowerCase() === location.name.toLowerCase() && azure.getResourceGroupFromId(virtualMachine.id).toLowerCase() === resourceGroup.name.toLowerCase()) {
+				let virtualMachineValue: CategoryValue;
+
+				// 1) check if VM is on by querying underlying compute resource's instance view
+				let vmInstanceView = await azure.getVMInstanceView(virtualMachine, account, subscription);
+				if (!vmInstanceView.statuses.some(status => status.code == "PowerState/running")) {
+					virtualMachineValue = {
+						name: virtualMachine.id,
+						displayName: constants.UNAVAILABLE_TARGET_PREFIX(virtualMachine.name)
+					}
+				}
+
+				// 2) check for IaaS extension in Full mode
+				else if (virtualMachine.properties.sqlManagement != "Full") {
+					virtualMachineValue = {
+						name: virtualMachine.id,
```

Comment:
Is there a value in exposing to CX why target is unavailable? Or do you think generic prefix is preferred?

Reply:
Generic prefix should be good enough, once the user selects it then the full error message will appear in the banner